app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/MenuActivity.cs

```diff
@@ -0,0 +1,261 @@
+            recyclerView.SetAdapter(adapterUserList);
+
+            InitialiseInputs();
+
+            AnimationExtension.AnimateBackground(FindViewById(Resource.Id.menuActivity));
+        }
+
+        public override void OnBackPressed()
+        {
+            DrawerLayout drawer = FindViewById<DrawerLayout>(Resource.Id.drawer_layout);
+            if (drawer.IsDrawerOpen(GravityCompat.Start))
+            {
+                drawer.CloseDrawer(GravityCompat.Start);
+            }
+            else
+            {
+                base.OnBackPressed();
+                Finish();
+            }
+        }
+
+        private void LogoutWindowShow()
+        {
+            var dialog = this.ConfirmationAlert(
+                    titleRes: Resource.String.logout,
+                    messageRes: Resource.String.logout_confirmation_message,
+                    positiveButtonRes: Resource.String.yes,
+                    negativeButtonRes: Resource.String.no,
+                    positiveCallback: delegate { HandleLogout(); });
+            dialog.Show();
+            dialog.Dispose();
+        }
+
+        private void HandleLogout()
+        {
+            StartActivity(new Intent(this, typeof(SignInActivity)));
+            FinishAffinity();
+        }
+
+        public bool OnNavigationItemSelected(IMenuItem item)
+        {
+            int id = item.ItemId;
+
+            if (id == Resource.Id.nav_groups)
+            {
+                StartActivity(new Intent(this, typeof(GroupActivity)));
+            }
+            else if (id == Resource.Id.nav_history)
+            {
+                StartActivity(new Intent(this, typeof(HistoryActivity)));
+            }
+            else if (id == Resource.Id.nav_menu_logout)
+            {
+                LogoutWindowShow();
+            }
+
+            Log.Info(TAG, "Interraction with navigation window");
+            DrawerLayout drawer = FindViewById<DrawerLayout>(Resource.Id.drawer_layout);
+            drawer.CloseDrawer(GravityCompat.Start);
+            return true;
+        }
+
+        private void InitialiseInputs()
+        {
+            var AddUserToGroup = FindViewById(Resource.Id.AddToGroup);
+            var CheckGroupHistory = FindViewById(Resource.Id.CheckGroupHistory);
+            var CheckUserHistory = FindViewById(Resource.Id.CheckPersonHistory);
+            var DisplayUsers = FindViewById(Resource.Id.DisplayYourGroupMembersButton);
+
+            AddUserToGroup.Click += delegate {
+                LayoutInflater layoutInflater = LayoutInflater.From(this);
+                View view = layoutInflater.Inflate(Resource.Layout.user_input_dialog_box, null);
+
+                var groupsList = GroupActivityModel.groups.Where(group => group.IsMember(SignInViewModel.currentUser.Username))
+                .Select(group => group.Groupname).ToList();
+
+                var spinner = view.FindViewById<Spinner>(Resource.Id.spinnerGroupName);
```

Comment:
you could have initialized this view outside the onclick method, now you are initializing it four times in code

Reply:
Views need to be reinitialized as they keep updating the spinner changes depending on the ammount of groups the user has. If he creates 1 or deletes 1 the spinner needs to be updated and this calls for view update.